### PR TITLE
front end changes for middle name and suffix capture

### PR DESCRIPTION
**Pull request by yhedholm**
Most of the changed lines come from the changes in the mock data file.

There are two trivial changes on the backend (fixing a spelling error and creating a parallel database for easy testing).

Apart from the mock data changes, I am just adding two extra name field, wherever name fields are handled.

### CODE REVIEW

File: frontend/rolecall/src/app/user/user-editor.component.html [hunk 1]
```diff
@@ -12,8 +12,12 @@ <h1 class="left-card-title">
...
             (click)="setCurrentUser({user : user, fromInputChange : false, shouldSetLastUser : true})">
           <h2 class="user-name">
-            {{ (user.first_name ? user.first_name : " ") + " " + (user.last_name ? user.last_name
-              : " ") }}
+            {{
+              (user.first_name ? user.first_name + " " : "") + 
+              (user.middle_name ? user.middle_name + " " : "") +
+              (user.last_name ? user.last_name : "") + 
+              (user.suffix ? user.suffix : "") 
+            }}
```
Comment (mikzat) [inline frontend/rolecall/src/app/user/user-editor.component.html:20]: Missing space before suffix.

File: frontend/rolecall/src/app/user/user-editor.component.scss [hunk 2]
```diff
@@ -1,5 +1,6 @@
 @import "src/colors";
 @import "src/constants";
+@import "src/std";
```
Comment (mikzat) [inline frontend/rolecall/src/app/user/user-editor.component.scss:3]: Is this actually used anywhere?

File: frontend/rolecall/src/app/user/user-editor.component.html [hunk 3]
```diff
@@ -12,8 +12,12 @@ <h1 class="left-card-title">
             [ngClass]="user.uuid == currentSelectedUser.uuid ? 'selected-user' : ''"
             (click)="setCurrentUser({user : user, fromInputChange : false, shouldSetLastUser : true})">
           <h2 class="user-name">
-            {{ (user.first_name ? user.first_name : " ") + " " + (user.last_name ? user.last_name
-              : " ") }}
+            {{
+              (user.first_name ? user.first_name + " " : "") +
```
Comment (mikzat) [inline frontend/rolecall/src/app/user/user-editor.component.html:16]: optional: would be nice to create a pure pipe that converts a user to a full name, then the same pipe can be used both in the template and in the class.

File: frontend/rolecall/src/app/performance/performance-editor.component.ts [hunk 4]
```diff
@@ -691,7 +691,10 @@ export class PerformanceEditor implements OnInit, OnDestroy, AfterViewChecked {
               groups: val.groups.map(g => {
                 return {
                   ...g,
-                  memberNames: g.members.map(mem => this.userAPI.users.get(mem.uuid)).map(usr => usr.first_name + " " + usr.last_name)
+                  memberNames: g.members.map(mem => this.userAPI.users.get(mem.uuid)).map(
+                      usr => usr.first_name + " " +
+                      (usr.middle_name.length > 0 ? usr.middle_name + " " : "") +
+                      usr.last_name + usr.suffix),
```
Comment (mikzat) [inline frontend/rolecall/src/app/performance/performance-editor.component.ts:697]: Missing space before suffix.

File: frontend/rolecall/src/app/user/user-editor.component.html [hunk 5]
```diff
@@ -48,13 +57,27 @@ <h1 class="right-card-title">
               [initValue]="(currentSelectedUser | emptyStringIfUndefined)['first_name'] | emptyStringIfUndefined">
           </app-text-input>
         </div>
+        <div class="name-input">
+          <app-text-input
+              valueName="Middle Name"
+              (valueChange)="onInputChange($event)"
+              [initValue]="(currentSelectedUser | emptyStringIfUndefined)['middle_name'] | emptyStringIfUndefined">
```
Comment (mikzat) [inline frontend/rolecall/src/app/user/user-editor.component.html:64]: This will produce invalid code if `currentSelectedUser` is actually undefined, because it will try to take a substring from an empty string.

The proper thing to do would be to use null-safe property access:
```
[initValue]="currentSelectedUser?.middle_name || ''"
```

**Review by mikzat — CHANGES_REQUESTED** [review]

File: frontend/rolecall/src/app/user/user-editor.component.html (hunk 5 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/user/user-editor.component.html:64]: Fixed.

File: frontend/rolecall/src/app/performance/performance-editor.component.ts (hunk 4 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/performance/performance-editor.component.ts:697]: See comment above.

File: frontend/rolecall/src/app/user/user-editor.component.html (hunk 3 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/user/user-editor.component.html:16]: I like the idea! Let's do that as part of a later cleanup.

File: frontend/rolecall/src/app/user/user-editor.component.scss (hunk 2 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/user/user-editor.component.scss:3]: Yes, this is where I put my standard style classes.

File: frontend/rolecall/src/app/user/user-editor.component.html (hunk 1 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/user/user-editor.component.html:20]: This is actually a 'feature'. The 'how to write a suffix' manuals are split on how to write John Smith junior. About half say it should be John Smith, Jr and the other half say it should be John Smith Jr. The easiest way to handle this is to give the user full control over the spacing. We discussed this with the Alving Ailey people and they liked this solution.

File: frontend/rolecall/src/app/user/user-editor.component.scss (hunk 2 above)
Comment (mikzat) [reply frontend/rolecall/src/app/user/user-editor.component.scss:3]: What I meant was I don't see anything from that file being used in `user-editor` styles, so it looks like it's imported unnecessarily?

Comment (yhedholm) [reply frontend/rolecall/src/app/user/user-editor.component.scss:3]: I was going include stc/std in most scss files so that they are hooked up properly as we put more code into std.scss.

File: frontend/rolecall/angular.json [hunk 6]
```diff
@@ -29,7 +29,8 @@
             ],
             "styles": [
               "./node_modules/@angular/material/prebuilt-themes/deeppurple-amber.css",
-              "src/styles.scss"
+              "src/styles.scss",
+              "src/std.scss"
```
Comment (mikzat) [inline frontend/rolecall/angular.json:33]: Revert the changes to this file and let's get https://github.com/googleinterns/role-call/pull/27 merged in.

**Review by mikzat — CHANGES_REQUESTED** [review]

File: frontend/rolecall/angular.json (hunk 6 above)
Comment (yhedholm) [reply frontend/rolecall/angular.json]: Done.

**Review by mikzat — APPROVED** [review]
